Library/Homebrew/utils/github/api.rb

```diff
@@ -5,7 +5,8 @@
 require "utils/shell"
 
 module GitHub
-  API_URL = "https://api.github.com"
+  PUBLIC_GITHUB_API_URL = "https://api.github.com"
+  API_URL = ENV.fetch("HOMEBREW_GITHUB_API_URL", PUBLIC_GITHUB_API_URL).freeze
```

Comment:
This will override e.g. `brew search` behaviour which seems undesirable.

Reply:
I'm game for searching internal taps but this would break searching non-internal taps.